internal/core/command/get_test.go

```diff
@@ -108,9 +116,99 @@ func TestNewGetCommandNoCorrelationIDInContext(t *testing.T) {
 	device := testDevice
+	req := newRequestWithHeaders(map[string]string{clients.ContentType: clients.ContentTypeCBOR}, http.MethodGet)
 	device.Service.Addressable.Address = "!@#$"
-	_, err := NewGetCommand(device, testCommand, "", context.Background(), nil, logger.NewMockClient())
+	_, err := NewGetCommand(device, testCommand, context.Background(), nil, logger.NewMockClient(), req)
 	if err == nil {
 		t.Errorf("The invalid URL error was not properly propagated to the caller")
 	}
 }
+
+func TestNewGetCommandContentType(t *testing.T) {
+	tests := []struct {
+		name            string
+		expectedHeaders map[string]string
+		req             *http.Request
+	}{
+		{
+			"cbor content type header propagated",
+			map[string]string{
+				clients.ContentType: clients.ContentTypeCBOR},
+			newRequestWithHeaders(map[string]string{
+				clients.ContentType: clients.ContentTypeCBOR}, http.MethodGet),
+		},
+		{
+			"json content type header propagated",
+			map[string]string{
+				clients.ContentType: clients.ContentTypeJSON},
+			newRequestWithHeaders(map[string]string{
+				clients.ContentType: clients.ContentTypeJSON}, http.MethodGet),
+		},
+		{
+			"no content type header provided",
+			map[string]string{
+				clients.ContentType: ""},
+			newRequestWithHeaders(map[string]string{
+				clients.ContentType: ""}, http.MethodGet),
+		},
+		{
+			"user-agent header not propagated",
+			map[string]string{
+				"User-Agent": ""},
+			newRequestWithHeaders(map[string]string{
+				"User-Agent": "User-Agent",
+			}, http.MethodGet),
+		},
+		{
+			"content-md5 header not propagated",
+			map[string]string{
+				"Content-MD5": ""},
+			newRequestWithHeaders(map[string]string{
+				"Content-MD5": "Content-MD5",
+			}, ""),
+		},
+		{
+			"json content type propagated, content-md5 header not propagated",
+			map[string]string{
+				clients.ContentType: clients.ContentTypeJSON,
+				"Content-MD5":       ""},
+			newRequestWithHeaders(map[string]string{
+				clients.ContentType: clients.ContentTypeJSON,
+				"Content-MD5":       "Content-MD5"}, http.MethodGet),
+		},
+		{
+			"cbor content type propagated, user-agent header not propagated, content-md5 header not propagated",
+			map[string]string{
+				clients.ContentType: clients.ContentTypeCBOR,
+				"User-Agent":        "",
+				"Content-MD5":       "",
+			},
+			newRequestWithHeaders(map[string]string{
+				clients.ContentType: clients.ContentTypeCBOR,
+				"User-Agent":        "User-Agent",
+				"Content-MD5":       "Content-MD5",
+			}, http.MethodGet),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var loggerMock = logger.NewMockClient()
+			ctx := context.Background()
+			getCommand, _ := NewGetCommand(
+				testDevice,
+				testCommand,
+				ctx,
+				nil,
+				loggerMock,
+				tt.req)
+			actualHeaders := map[string]string{}
+			for k := range tt.expectedHeaders {
+				actualHeaders[k] = getCommand.(serviceCommand).Request.Header.Get(k)
+			}
```

Comment:
This looks like we are not capturing all the `actualHeaders`. We are only getting the header values specified in the `expectedHeaders` Should we get all the headers from `getCommand` and inspect that the expected values are all present and not additional ones are added?

Reply:
My apologies for the confusing way this was set up... I have fixed it.